File: src/org/webfabric/sitemesh/OriginalPathInfo.java
package org.webfabric.sitemesh;

import javax.servlet.http.HttpServletRequest;

class OriginalPathInfo implements Path {
    private final String value;

    public OriginalPathInfo(HttpServletRequest request) {
        this.value = (String) request.getAttribute("javax.servlet.include.servlet_info");
    }

    public OriginalPathInfo(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }

}

File: src/org/webfabric/servlet/OriginalPathInfo.scala
package org.webfabric.servlet;

import org.webfabric.io.{HierarchicalPath}
import javax.servlet.http.HttpServletRequest;

class OriginalPathInfo(v: String) extends HierarchicalPath(v)

object OriginalPathInfo {
  def apply(request: HttpServletRequest):OriginalPathInfo = {
    return new OriginalPathInfo(request.getAttribute("javax.servlet.include.servlet_info").asInstanceOf[String]);
  }
}
